torch/csrc/autograd/saved_variable.cpp

```diff
@@ -55,6 +51,24 @@ Variable SavedVariable::unpack(std::shared_ptr<Function> saved_for) const {
     grad_fn = std::move(saved_for);
   }
 
+  if (saved_version_ != version_counter_.current_version()) {
+    std::stringstream message;
+    message << "one of the variables needed for gradient computation has been "
+        "modified by an inplace operation: [" << data_.type().toString() << " "
+        << data_.sizes() << "]";
+    if (grad_fn) {
+        message << ", which is output " << output_nr_
+            << " of " << grad_fn->name() << ",";
+    }
+    message << " is at version " << version_counter_.current_version()
+        << ", not expected version " << saved_version_ << ".";
+    if (!AnomalyMode::is_enabled()) {
+        message << " Hint: enable anomaly detection to find the "
+            << "forward pass operation which modified it.";
```

Comment:
Is it going to point to the forward pass op that modified it? Or the one that needs it?

Reply:
Good spot. From what I read in https://github.com/pytorch/pytorch/issues/15803#issuecomment-454030272, I assumed that it would find the inplace operation, but after debugging my problem, I found that this was a red herring. I will change the hint accordingly!

Reply:
It would be quite interesting to actually be able to point out both for debugging purposes !
But it would be a bit tricky to do :/

Reply:
Well, where do the version counters live? Would it be possible to have a mode that warns if any gradient-required tensor is involved when increasing a version counter? Or a mode that simply warns on any in-place operation?

Reply:
There are usually many (many) inplace ops. So that mode might be too verbose to be useful.
But maybe it's something we want to add to the anomaly detection mode: track which python line made the version counter bump so that we can use that information here for better error printing.
But that would be a much more involved change.